### PR TITLE
LB-583: Add tests for Spotify search functionality

### PR DESCRIPTION
**Pull request by vardansaini**
# Problem

Spotify search was not being tested
https://tickets.metabrainz.org/browse/LB-583

# Solution

Added test for Spotify Search

# Action

in utils.test I added tests using Jest for Spotify search and I added a JSON file for response

### CODE REVIEW

**Review by MonkeyDo — COMMENTED** [review]
That's a great start, thanks vardan !

I'd also like to see some tests for when things go bad.
For example, what if the `fetch` call itself fails?
And what if the call succeeds, but Spotify API returns an error message?
And what if everything succeeds but there are no matching results?

File: listenbrainz/webserver/static/js/tests/utils/utils.test.tsx [hunk 1]
```diff
@@ -99,3 +105,37 @@ describe("preciseTimestamp", () => {
...
+        method: "GET",
+        headers: {
+          Authorization: "Bearer foobar",
+          "Content-Type": "application/json",
+        },
+      }
+    );
+  });
+
+  it("we retrieve first track from json", async () => {
```
[the remark this answers is not in the document]
Comment (vardansaini) [reply listenbrainz/webserver/static/js/tests/utils/utils.test.tsx:136]: Got it!

File: listenbrainz/webserver/static/js/tests/utils/utils.test.tsx [hunk 2]
```diff
@@ -99,3 +105,37 @@ describe("preciseTimestamp", () => {
...
+  beforeEach(() => {
+    // Mock function for fetch
+    window.fetch = jest.fn().mockImplementation(() => {
+      return Promise.resolve({
+        ok: true,
+        status: 200,
+        json: () => Promise.resolve(spotifySearchResponse),
+      });
+    });
+    jest.useFakeTimers();
```
[the remark this answers is not in the document]
Comment (vardansaini) [reply listenbrainz/webserver/static/js/tests/utils/utils.test.tsx:119]: fixed!

File: listenbrainz/webserver/static/js/tests/utils/utils.test.tsx [hunk 3]
```diff
@@ -137,4 +138,42 @@ describe("searchForSpotifyTrack", () => {
...
+    window.fetch = jest.fn().mockImplementation(() => {
+      // 1st call will recieve a network error
+      return Promise.reject(new Error("Oh no!"));
+    });
+    await expect(
+      searchForSpotifyTrack("foobar", "import", "vs", "star")
+    ).rejects.toThrow(Error("Oh no!"));
+  });
+
+  it("skips if any other response code is recieved", async () => {
```
Comment (vardansaini) [inline listenbrainz/webserver/static/js/tests/utils/utils.test.tsx:153]: I think the code in utils.tsx is wrong or has some issues at line number L: 46 to 48, it is trying to throw string directly which is probably wrong. That's why this test is failing. Please review this!

Comment (MonkeyDo) [reply listenbrainz/webserver/static/js/tests/utils/utils.test.tsx:153]: You're right @vardansaini , it should be `throw new Error(responseBody.error);`. Can you please make the required modification?

Comment (vardansaini) [reply listenbrainz/webserver/static/js/tests/utils/utils.test.tsx:153]: sure!

Comment (vardansaini) [reply listenbrainz/webserver/static/js/tests/utils/utils.test.tsx:153]: done

**Review by MonkeyDo — APPROVED** [review]
Thank you very much for  working on this !
Always great when tests are added :)

**Pull request merged**